### DISCUSSION TITLE
A potential bug in the Misc of CreatingArduino node?!

### DISCUSSION THREAD

**Discussion by chiyu1203** [bonsai-rx/bonsai#1384] [Q&A]
Dear Bonsai team,
I used the example workflow from Bonsai tutorial [Data Synchronization exercise 2](https://bonsai-rx.org/docs/tutorials/synching.html) and set the Name of CreatingArduino node, MyArduino, and PortName to be COM3. However, when I configured the Portname in DigitalOutput (or any other nodes from the Arduino library), the only name I could choose is "MyArduino". Then I ran this workflow one time and it was fine. However, when I tried to run this workflow again, I saw the following error saying that my PortName in the DigitalOutput does not start from "COM". At the moment, I can work around this error is to either set the name of CreatingArduino to be COM3 so that the two DigitalOutput can use that "COM" name as the Portname; or I leave the name of CreatingArduino blank so that the DigitalOutput is forced only choose from CreatingArduino's PortName. However, I was curious if this is a potential bug in their Misc?!
Many thanks in advance!
[ttl_pulse.zip](https://github.com/bonsai-rx/bonsai/files/11548336/ttl_pulse.zip)

![image](https://github.com/bonsai-rx/bonsai/assets/16696523/de3d394c-3142-4a54-b27e-097e518d489f)

**Comment by bruno-f-cruz** [comment]
Move the CreateArduino to the top of the workflow, if I am understanding the problem correctly, that should fixe it or throw a message stating that the COM port is not available if such is the case.

**Reply by chiyu1203** [reply]
Hi, I just moved the CreateArduino to the top of the workflow and it did not fix the problem....I shall also say that I only see that error messages when I put two DigitalOutput nodes in the workflow (if I remove one of the nodes, I can run that workflow without any error, even though the PortName does not start from COM)

**Reply by bruno-f-cruz** [reply]
I just had a look at your workflow and i am not sure I follow why one of the DigitalOutputs targets COM3 and the other one MyArduino. Should they not both target MyArduino?

**Reply by chiyu1203** [reply]
Sorry for the confusion! They both targeted on MyArduino in the beginning but I was playing around different PortName and I forgot to change the name back....
